### PR TITLE
Counts and Lists: Improvements to SDK docs

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup

/kind documentation

> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

Much like the player tracking SDK documentation, I wanted to be explicit in the SDK documentation for Counts and Lists where data was being stored, and where default values where coming from with links back to the generated API reference - so made some improvements to help facilitate that.

**Which issue(s) this PR fixes**:

Work on #2716

**Special notes for your reviewer**:

This does still need a Counters and Lists landing page to reference (this is my next task) as well, but wanted to get this done while I was thinking about it.

### CODE REVIEW

File: site/content/en/docs/Guides/Client SDKs/_index.md [hunk 1]
```diff
@@ -243,80 +249,94 @@ SDK Server sidecar process.
...
 ##### Alpha().GetListCapacity(key)
 
-This function retrieves the capacity of a specified list.
+This function retrieves either the [`GameServer.Status.Lists[key].Capacity`][gameserverstatus] or the SDK
+awaiting-batch value for the given key, whichever is most up to date.
+
+##### Alpha().GetListValues(key)
+
+This function retrieves either the [`GameServer.Status.Lists[key].Values`][gameserverstatus] or the SDK
+awaiting-batch value for the given key, whichever is most up to date.
```
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/_index.md:327]:
```suggestion
awaiting-batch values array for the given key, whichever is most up to date.
```

File: site/content/en/docs/Guides/Client SDKs/_index.md [hunk 2]
```diff
@@ -243,80 +249,94 @@ SDK Server sidecar process.
...
-This function returns a boolean value on if a specific string value exists in a list's values. 
+Convenience function, which returns if the specific string value exists in the results
+of [`Alpha().GetListValues(key)`](#alphagetlistvalueskey).
 
 ##### Alpha().GetListLength(key)
 
-This function retrieves the number of items (length) in a  specified List.
-
-##### Alpha().GetListValues(key)
+Convenience function, which retrieves the length the results of [`Alpha().GetListValues(key)`](#alphagetlistvalueskey).
```
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/_index.md:336]:
```suggestion
Convenience function, which retrieves the length of the results of [`Alpha().GetListValues(key)`](#alphagetlistvalueskey).
```

**Review by igooch — CHANGES_REQUESTED** [review]

File: site/content/en/docs/Guides/Client SDKs/_index.md [hunk 3]
```diff
@@ -243,80 +249,94 @@ SDK Server sidecar process.
...
 
-All functions will return an error if the specified `key` is not predefined in the `GameServer.spec.counters` resource
-configuration.
+All functions will return an error if the specified `key` is not predefined in the 
+[`GameServer.Spec.Counters`][gameserverspec] resource configuration.
 
 ##### Alpha().GetCounterCount(key)
 
-This function retrieves the current count for a counter in the game server.
+This function retrieves either the [`GameServer.Status.Counts[key].Count`][gameserverstatus] or the SDK awaiting-batch
```
[the remark this answers is not in the document]
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/_index.md:265]: Blah, thank you! I'm going to break that habit eventually.

File: site/content/en/docs/Guides/Client SDKs/_index.md [hunk 4]
```diff
@@ -243,80 +249,94 @@ SDK Server sidecar process.
 However, changes made through Allocation or the Kubernetes API to `GameServer` List and Counter values will be 
 eventually consistent when being retrieved through the SDK.
 
-Since the SDK update operations back to the `GameServer.status` values is batched asynchronous, it is worth noting that
-any value incremented past a counter or list capacity may be silently truncated to the currently set capacity if 
-modified concurrently through the SDK and Allocation/Kubernetes API.
+Since the SDK update operations back to the 
+[`GameServer.status`]({{< ref "/docs/Reference/agones_crd_api_reference.html#agones.dev/v1.GameServerStatus" >}}) values
+is batched asynchronously, it is worth noting that any value incremented past a counter or list capacity may be silently
+truncated to the currently set capacity if modified concurrently through the SDK and Allocation/Kubernetes API.
```
[the remark this answers is not in the document]
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/_index.md:255]: Took a stab at a rewrite, I think it reads better, but PTAL and let me know what you think!

[the remark this answers is not in the document]
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/_index.md]: Just bumping to see if this update makes more sense?

File: site/content/en/docs/Guides/Client SDKs/_index.md [hunk 5]
```diff
@@ -243,80 +249,94 @@ SDK Server sidecar process.
...
 
-This function retrieves the current count for a counter in the game server.
+This function retrieves either the [`GameServer.Status.Counts[key].Count`][gameserverstatus] or the SDK awaiting-batch
+value for a given key, whichever is most up to date.
 
 ##### Alpha().SetCounterCount(key, amount)
 
-This function sets the count to a given value. This operation overwrites any previous values and the new value cannot
-exceed the Counter's capacity.
+This function sets the value of [`GameServer.Status.Counts[key].Count`][gameserverstatus] for the given key to the
```
[the remark this answers is not in the document]
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/_index.md]: OMG. I didn't push these changes 🤦🏻

**Comment by markmandel** [comment]
Didn't push the changes last week 🤦🏻 that should be better now.

File: site/content/en/docs/Guides/Client SDKs/_index.md [hunk 6]
```diff
@@ -230,93 +230,115 @@ observable through the Kubernetes API.
...
+will be eventually consistent when being retrieved through the SDK.
 
-Since the SDK update operations back to the `GameServer.status` values is batched asynchronous, it is worth noting that
-any value incremented past a counter or list capacity may be silently truncated to the currently set capacity if 
-modified concurrently through the SDK and Allocation/Kubernetes API.
-{{< /alert >}}
+In the event where a Counter or List may extend past the maximum capacity Allocation/API changes to
+[`GameServer.status`]({{< ref "/docs/Reference/agones_crd_api_reference.html#agones.dev/v1.GameServerStatus" >}}) values
+while a batched Counter or List modification operation has yet to be applied, those values will be silently truncated to
+the capacity of that Counter or List.
```
Comment (igooch) [inline site/content/en/docs/Guides/Client SDKs/_index.md:256]: This is still confusing. Subject -> Verb -> Object is an easier format to understand. 

```suggestion
The Agones SDK server batches the update operations of [`GameServer.Status.Counters` and `GameServer.Status.Lists`][gameserverstatus] asynchronously. This means that if you update [`GameServer.status`]({{< ref "/docs/Reference/agones_crd_api_reference.html#agones.dev/v1.GameServerStatus" >}}) values through both the SDK and the Allocation/Kubernetes API, the batch processing may silently truncate some of those values to the capacity of that Counter or List.
```

**Review by igooch — APPROVED** [review]
LGTM, just one comment on clarity.

File: site/content/en/docs/Guides/Client SDKs/_index.md (hunk 6 above)
Comment (markmandel) [reply site/content/en/docs/Guides/Client SDKs/_index.md:256]: Thank you! That does read much better. Adding change and updating!

**Pull request merged**